android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -175,6 +175,16 @@ protected boolean updateSelectionRange(int selStart, int selEnd) {
       String rawText = icText.text.toString();
       updateText(rawText.toString());
 
+      int selStart = icText.selectionStart;
+      int selEnd = icText.selectionEnd;
```

Comment:
By getting the selection-state information from the same source and operation as we get the current text, we can avoid desynchronization here.